cmake/compiler/gcc/target_nostdinc.cmake

```diff
@@ -0,0 +1,8 @@
+
+macro(toolchain_cc_nostdinc)
+
+  zephyr_compile_options(
+    -nostdinc
```

Comment:
I'm fine with this. 
But if we are to follow precedence of the other toolchain abstractions, this should set a variable, and leave the call to `zephyr_compile_options` to the root CMakeLists.txt.